src/main/java/io/r2dbc/postgresql/TargetServerType.java

```diff
@@ -0,0 +1,53 @@
+package io.r2dbc.postgresql;
+
+import javax.annotation.Nullable;
+
+public enum TargetServerType {
```

Comment:
Should we convert `TargetServerType` to an `abstract class`/`interface` + default `enum` so that users can implement their own logic into `allowStatus`?